CMakeLists.txt

```diff
@@ -1,83 +1,191 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (C) The BioDynaMo Project.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# See the LICENSE file distributed with this work for details.
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# -----------------------------------------------------------------------------
+
 cmake_minimum_required(VERSION 3.2.0)
+cmake_policy(VERSION 3.3.0)
 # With CMAKE >= 3.2 external projects support byproducts, this is needed
 # to support Ninja as generator
 
-project(BioDynaMo)
-set(BDM_VERSION 0.1.0)
+#---Set the locale to default C to prevent issued due to localization of commands---------------
+# This is necessary as we for example call `clang -v` and parse its output. But on a localized
+# program, the output parsing is much more error prone as certrain strings we're looking for
+# could be missing or be in a different order. To prevent those errors, let's just force all
+# output to use the default C locale which is more or less identical on all systems.
+set(ENV{LANG} C)
+
+# Project name and current version. We do not specify any language for
+# this project. Therefore, we will be able to check manually which compiler
+# we detected on the system.
+project(BioDynaMo LANGUAGES VERSION "0.1.0")
+
+# Directory in which we saved all the cmake informations.
+set(ENV{BDM_CMAKE_DIR} ${CMAKE_SOURCE_DIR}/cmake)
+
+# Add an additional path to make include() look in the right places.
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "$ENV{BDM_CMAKE_DIR}")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/third_party/cmake/")
+
+# Include some cmake files with utilities
+include(external/SHA256Digests)
+include(FeatureSummary)
+include(utils)
+
+# Detect os type and version
+if (NOT DEFINED OS)
+    SET(DETECTED_OS "none")
+    detect_os()
+else()
+    SET(DETECTED_OS ${OS})
+    check_detected_os(${DETECTED_OS})
+endif()
+
+# Find a suitable compiler
+include(CheckCompiler)
+
+# Directory in which BioDynaMo will be installed. It can be provided
+# by the user itself.
+if (NOT DEFINED ${BDM_INSTALL_DIR})
+    set(BDM_INSTALL_DIR $ENV{HOME}/.bdm)
+endif()
+
+# Specify some directories which will be used to structure BioDynaMo build.
+set(CMAKE_BIODYNAMO_ROOT        "${CMAKE_BINARY_DIR}")
```

Comment:
`CMAKE_BUILD_ROOT` would be more intuitive

Reply:
I have opted for `CMAKE_BIODYNAMO_BUILD_ROOT`. `CMAKE_BUILD_ROOT` seems to be related to the ROOT framework.